File: 1133_largest_unique_number.py
from typing import List


class Solution:
    def largestUniqueNumber(self, nums: List[int]) -> int:
        map = {}
        for num in nums:
            if num in map:
                map[num] += 1
            else:
                map[num] = 1

        mx = -1
        for (key, val) in map.items():
            if (val == 1):
                mx = max(mx, key)
        
        return mx

File: 1133_largest_unique_number.cpp
#include <unordered_map>
#include <vector>

class Solution {
   public:
    int largestUniqueNumber(std::vector<int> &nums) {
        int mx = -1;

        std::unordered_map<int, int> map;
        for (auto &num : nums) ++map[num];

        for (auto &[key, val] : map)
            if (val == 1) mx = std::max(mx, key);

        return mx;
    }
};